### PR TITLE
NCR Jobs, Dogtags & Medallions, Radiopack fix

### PR DESCRIPTION
**Pull request by bardinatavern4**
Adds dogtags, medals, cleans up some of the NCR code and sets backpack/boots/gloves to be universal as well as the  forbids and enforces sections when you spawn in. Finally, finishes the NCR jobs.

### CODE REVIEW

**Comment by bardinatavern4** [comment]
Additions:

Projectile wound values tested by me and Hekzder. Seems to be in a good spot for lethal but not too bullshit combat, might need further adjustment in the future.
Added recruit IDs for both Legion and NCR. A lieutenant or centurion can use their ID on their factions recruit tags to name them, letting them recruit wastelanders and give them IDs.

**Review by uomo91 — CHANGES_REQUESTED** [review]

File: mojave/items/dogtags.dm [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+/obj/item/card/id/ms13
+	name = "ID Tags"
+	desc = "A simple identification tag.."
+	icon = 'mojave/icons/objects/dogtags.dmi'
+	icon_state = "holodogtag"
+	uses_overlays = FALSE
+	var/datum/bank_account = null
+
+/obj/item/card/id/ms13/dogtag
```
Comment (uomo91) [inline mojave/items/dogtags.dm:14]:
```suggestion
/obj/item/card/id/ms13/bos
```

File: mojave/items/dogtags.dm [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	return ..()
+
+/obj/item/card/id/ms13/sheriff
+	name = "sheriff's badge"
+	desc = "A golden Sheriff's badge."
+	icon_state = "sheriff"
+
+/obj/item/card/id/ms13/ncrrecruit
+	name = "recruit's tags"
+	desc = "A dog tag proving enlistment. Issued to local recruits."
```
Comment (uomo91) [inline mojave/items/dogtags.dm:39]:
```suggestion
	desc = "A dog tag proving enlistment in the NCR, issued to local recruits."
```

File: mojave/items/dogtags.dm [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+/obj/item/card/id/ms13/ncrrecruit/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/card/id/ms13/ncrlieutenant))
+		registered_name = stripped_input(user, "Who do you want to designate as a recruit?", , "", MAX_NAME_LEN)
+		to_chat(user, "You scribble the [registered_name] for the name on the dogtag.")
+		update_label()
+	return ..()
+
+/obj/item/card/id/ms13/ncrtrooper
+	name = "trooper's tags"
+	desc = "A dog tag proving enlistment."
```
Comment (uomo91) [inline mojave/items/dogtags.dm:52]:
```suggestion
	desc = "A dog tag proving enlistment in the NCR."
```

File: mojave/items/dogtags.dm [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+	icon_state = "ncrdogtagtrooper"
+
+/obj/item/card/id/ms13/ncrsergeant
+	name = "sergeant's tags"
+	desc = "A chevron decorated dog tag showing NCO-ship."
```
Comment (uomo91) [inline mojave/items/dogtags.dm:57]:
```suggestion
	desc = "A chevron decorated dog tag showing NCO status within the NCR."
```

File: mojave/items/dogtags.dm [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+	icon_state = "ncrdogtagofficer"
+
+/obj/item/card/id/ms13/ncrcaptain
+	name = "captain's tags"
+	desc = "A dog tag that demands respect from all those subordinate to it. This one belongs to an NCR captain."
+	icon_state = "ncrdogtagcaptain"
+
+/obj/item/card/id/ms13/ncrranger
+	name = "ranger tags"
+	desc = "A dog tag that invokes fear in those who see it, belongs to a man with a big iron on his hip."
```
Comment (uomo91) [inline mojave/items/dogtags.dm:72]:
```suggestion
	desc = "A dog tag that invokes fear in those who see it. Belongs to an elite of the NCR, usually with a big iron on their hip."
```

File: mojave/items/dogtags.dm [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+	icon_state = "legionmedallionprime"
+
+/obj/item/card/id/ms13/legveteran
+	name = "veteran medallion"
+	desc = "A heavily marked silver disc stamped with the Legion's Bull insignia. Belongs to a veteran, and reeks of iron."
+	icon_state = "legionmedallionveteran"
+
+/obj/item/card/id/ms13/legcenturion
+	name = "centurion medallion"
+	desc = "A golden disc awarded to the most fierce men in the whole legion. If you are close enough to read the insignia you won't be alive much longer."
```
Comment (uomo91) [inline mojave/items/dogtags.dm:101]:
```suggestion
	desc = "A golden disc awarded to the most fierce men in the entire Legion. If you are close enough to read the insignia you won't be alive much longer."
```

File: mojave/jobs/job_types/ncr/lieutenant.dm [hunk 7]
```diff
@@ -2,10 +2,8 @@
 	title = "NCR Lieutenant"
 	total_positions = 1
 	spawn_positions = 1
-	supervisors = ""
-	description = ""
-	forbid = ""
-	enforce = ""
+	supervisors = "None, outside of exceptional circumstances."
+	description = "Organize the sergeants, corporals, troopers, and local recruits to secure the area."
```
Comment (uomo91) [inline mojave/jobs/job_types/ncr/lieutenant.dm:6]:
```suggestion
	description = "Organize and command your forces."
```

File: mojave/jobs/job_types/ncr/trooper.dm [hunk 8]
```diff
@@ -4,23 +4,31 @@
 	spawn_positions = 10
 	supervisors = "Lieutenant, Sergeant, and Corporal, in that order."
```
Comment (uomo91) [inline mojave/jobs/job_types/ncr/trooper.dm:5]:
```suggestion
	supervisors = "the Lieutenant, Sergeants, and Corporals, in that order."
```

File: mojave/jobs/job_types/ncr/sergeant.dm [hunk 9]
```diff
@@ -4,18 +4,28 @@
 	spawn_positions = 2
 	supervisors = "The Lieutenant"
```
Comment (uomo91) [inline mojave/jobs/job_types/ncr/sergeant.dm:5]:
```suggestion
	supervisors = "the Lieutenant"
```

File: mojave/jobs/job_types/ncr/radioman.dm [hunk 10]
```diff
@@ -3,27 +3,30 @@
 	total_positions = 2
 	spawn_positions = 2
 	supervisors = "the Corporals, the Sergeants, and the Lieutenant"
```
Comment (uomo91) [inline mojave/jobs/job_types/ncr/radioman.dm:5]:
```suggestion
	supervisors = "the Lieutenant, Sergeants, and Corporals. In that order."
```

File: mojave/jobs/job_types/ncr/corporal.dm [hunk 11]
```diff
@@ -4,8 +4,6 @@
 	spawn_positions = 4
 	supervisors = "Sergeant, and Lieutenant"
```
Comment (uomo91) [inline mojave/jobs/job_types/ncr/corporal.dm:5]:
```suggestion
	supervisors = "the Lieutenant and the Sergeants, in that order."
```

**Review by uomo91 — CHANGES_REQUESTED** [review]

**Review by uomo91 — APPROVED** [review]

File: mojave/items/dogtags.dm [hunk 12]
```diff
@@ -0,0 +1,102 @@
+/* The old dogtags
+* and medallions from BD.
+* all have the dogtag subtype
+*/
+
+/obj/item/card/id/ms13
+	name = "ID Tags"
+	desc = "A simple identification tag.."
```
Comment (uomo91) [inline mojave/items/dogtags.dm:8]:
```suggestion
	desc = "A simple identification tag. This is a base class and you shouldn't be seeing it."
```

File: mojave/items/dogtags.dm [hunk 13]
```diff
@@ -0,0 +1,102 @@
+/* The old dogtags
+* and medallions from BD.
+* all have the dogtag subtype
+*/
+
+/obj/item/card/id/ms13
+	name = "ID Tags"
```
Comment (uomo91) [inline mojave/items/dogtags.dm:7]:
```suggestion
	name = "\improper ID tag"
```

File: mojave/items/dogtags.dm [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+	desc = "A silver bar dog tag that denotes a member of the NCR military with a lieutenant commission."
+	icon_state = "ncrdogtagofficer"
+
+/obj/item/card/id/ms13/ncrcaptain
+	name = "captain's tags"
+	desc = "A dog tag that demands respect from all those subordinate to it. This one belongs to an NCR captain."
+	icon_state = "ncrdogtagcaptain"
+
+/obj/item/card/id/ms13/ncrranger
+	name = "ranger tags"
```
Comment (uomo91) [inline mojave/items/dogtags.dm:71]:
```suggestion
	name = "ranger's tags"
```

**Review by uomo91 — CHANGES_REQUESTED** [review]

**Review by uomo91 — APPROVED** [review]

File: mojave/jobs/job_types/ncr/medicalofficer.dm [hunk 15]
```diff
@@ -0,0 +1,35 @@
+/datum/job/ms13/ncr/medicalofficer
+	title = "NCR Medical Officer"
+	total_positions = 1
+	spawn_positions = 1
+	supervisors = "The Lieutenant."
+	description = "ou are in charge of providing medical support to the NCR throughout the region. You are a non-combat role and should not be using your authority for anything outside of medical purposes."
```
Comment (Hekzder) [inline mojave/jobs/job_types/ncr/medicalofficer.dm:6]:
```suggestion
	description = "You are in charge of providing medical support to the NCR throughout the region. You are a non-combat role and should not be using your authority for anything outside of medical purposes."
```

File: mojave/jobs/job_types/ncr/engineer.dm [hunk 16]
```diff
@@ -0,0 +1,32 @@
+/datum/job/ms13/ncr/engineer
+	title = "NCR Engineer"
+	total_positions = 1
+	spawn_positions = 1
+	supervisors = "the Lieutenant and the Sergeants, in that order."
```
Comment (Hekzder) [inline mojave/jobs/job_types/ncr/engineer.dm:5]:
```suggestion
	supervisors = "the Lieutenant, Sergeants, and Corporals.  In that order."
```

**Review by Hekzder — CHANGES_REQUESTED** [review]
Small stuff. One is a missing letter thing, other is a chain of command fix. E-4 Specialists are still technically subordinate to E-4 Corporals. One is a, well, specialist in their field, other is an NCO.

**Review by Hekzder — APPROVED** [review]
Should be good now. Testing then will merge based on results.

File: mojave/items/storage/radiopack.dm [hunk 17]
```diff
@@ -1,4 +1,4 @@
-/obj/item/ms13/radiopack
+/obj/item/ms13/storage/backpack/radiopack
```
Comment (Hekzder) [inline mojave/items/storage/radiopack.dm:1]: Pathing should be changed for this, ideally with the other backpacks as well if you're willing.

File: mojave/items/dogtags.dm [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+	icon_state = "ncrdogtagtrooper"
+
+/obj/item/card/id/ms13/ncrrecruit/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/card/id/ms13/ncrlieutenant))
+		registered_name = stripped_input(user, "Who do you want to designate as a recruit?", , "", MAX_NAME_LEN)
+		to_chat(user, "You scribble the [registered_name] for the name on the dogtag.")
+		update_label()
+	return ..()
+
+/obj/item/card/id/ms13/ncrtrooper
```
Comment (Hekzder) [inline mojave/items/dogtags.dm:50]: Dogtags get the in-game name of ID cards for some reason. Not a super pressing bug that will hold up a merge, but worth noting here.

File: mojave/items/storage/radiopack.dm [hunk 19]
```diff
@@ -114,8 +130,22 @@
...
+		snap_back()
+
+/obj/item/radio/ms13/NCR/proc/snap_back()
+	if(!radiopack)
+		return
+	forceMove(radiopack)
+
+/obj/item/radio/ms13/NCR/doMove(atom/destination)
+	if(destination && (destination != radiopack.loc || !ismob(destination)))
+		if (loc != radiopack)
```
Comment (Hekzder) [inline mojave/items/storage/radiopack.dm:148]: This proc will probably need to be adjusted to not spam chat whenever you take it out or put it back in. Also not super pressing and will not delay a merge, but worth noting for future.

**Review by Hekzder — CHANGES_REQUESTED** [review]

File: mojave/items/storage/radiopack.dm (hunk 19 above)
Comment (uomo91) [reply mojave/items/storage/radiopack.dm:148]: He could just remove the to_chat if it's a problem.

File: mojave/items/dogtags.dm (hunk 18 above)
Comment (uomo91) [reply mojave/items/dogtags.dm:50]: I suspect they're getting a name from some base code proc.

**Pull request merged**